lib/monocle.js

```diff
@@ -175,6 +157,114 @@ var _handle = function(method, path, options) {
     return promise;
 };
 
+/**
+ * Returns true if the provided resource fullfills all of the requirements of the options.
+ *
+ * @param object resource - The resource to check
+ * @param object options - Request options
+ * @return boolean
+ */
+var isResourceComplete = function(resource, props) {
+    if (!props) {
+        return true;
+    }
+
+    for (var i in props) {
+        if (!props.hasOwnProperty(i)) {
+            continue;
+        }
+
+        if (!hasProp(resource, props[i])) {
```

Comment:
Since ```hasProp()``` is defined after this function, there could be a race condition in which the anonymous function will return undefined.  It may be better to put ```hasProp()``` before ```isResourceComplete()```, or just make ```hasProp()``` a function declaration instead.